libs/i18n/src/lib/pipes/fd-translate.pipe.ts

```diff
@@ -0,0 +1,133 @@
+import { ChangeDetectorRef, Inject, isDevMode, OnDestroy, Pipe, PipeTransform } from '@angular/core';
+import { get } from 'lodash-es';
+import {
+    BehaviorSubject,
+    combineLatest,
+    distinctUntilChanged,
+    filter,
+    map,
+    Observable,
+    skip,
+    Subject,
+    takeUntil
+} from 'rxjs';
+
+import { FD_LANGUAGE_ENGLISH } from '../languages';
+import { FdLanguage, FdLanguageKeyArgs, FdLanguageKey, FdLanguageKeyFunction } from '../models/lang';
+import { FD_LANGUAGE } from '../utils/tokens';
+
+@Pipe({
+    name: 'fdTranslate',
+    pure: false // required to update the value when the observable is resolved
+})
+export class FdTranslatePipe implements PipeTransform, OnDestroy {
+    /** @hidden */
+    private readonly _curlyBracesRegExp = /{{\s*([^{}\s]*)\s*}}/g;
+
+    /** @hidden */
+    private readonly _key$ = new BehaviorSubject<string | undefined>(undefined);
+
+    /** @hidden */
+    private readonly _args$ = new BehaviorSubject<FdLanguageKeyArgs | undefined>(undefined);
+
+    /** @hidden */
+    private _value: string | undefined;
+
+    /** @hidden */
+    private readonly _onDestroy$ = new Subject<void>();
+
+    constructor(@Inject(FD_LANGUAGE) private _language$: Observable<FdLanguage>, private _cdr: ChangeDetectorRef) {
+        this._instantiateSubscription();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+    }
+
+    transform(key: string, args?: FdLanguageKeyArgs, defaultValue = ''): string {
+        this._key$.next(key);
+        this._args$.next(args);
+
+        return this._value ?? defaultValue;
+    }
+
+    /** @hidden */
+    _interpolate(expression: string, args: FdLanguageKeyArgs = {}): string {
```

Comment:
Shouldn't this method be a private?

Reply:
@platon-rov well, it's left public for testing purposes. Generally it's better to avoid testing internals of a class, but this function is pretty important by itself